src/machine/asm/execute_x64.S

```diff
@@ -1251,6 +1254,278 @@ ckb_vm_x64_execute:
   WRITE_RD(RS1)
   NEXT_INST
 .p2align 3
+.CKB_VM_ASM_LABEL_OP_LR_W:
+  DECODE_R
+  movq REGISTER_ADDRESS(RS1), RS1
+  CHECK_READ_VERSION1(RS1, 4)
+  movslq CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_MEMORY(MACHINE, RS1), TEMP1
+  WRITE_RD(TEMP1)
+  movq RS1, LOAD_RESERVATION_ADDRESS
+  NEXT_INST
+.p2align 3
+.CKB_VM_ASM_LABEL_OP_SC_W:
+  DECODE_R
+  movq REGISTER_ADDRESS(RS1), RS1
+  movq REGISTER_ADDRESS(RS2r), RS2r
+  CHECK_WRITE(RS1, TEMP1d, 4)
+  movq LOAD_RESERVATION_ADDRESS, TEMP1
+  cmp RS1, TEMP1
+  jne .sc_w_branch
+  movq RS2r, CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_MEMORY(MACHINE, RS1)
+  movq $UINT64_MAX, LOAD_RESERVATION_ADDRESS
+  WRITE_RD_VALUE($0, TEMP1)
+  NEXT_INST
+.sc_w_branch:
+  WRITE_RD_VALUE($1, TEMP1)
+  NEXT_INST
+.p2align 3
+.CKB_VM_ASM_LABEL_OP_AMOSWAP_W:
+  DECODE_R
+  movq REGISTER_ADDRESS(RS1), RS1
+  movq REGISTER_ADDRESS(RS2r), RS2r
+  CHECK_READ_VERSION1(RS1, 4)
```

Comment:
I think we can skip `CHECK_READ_VERSION1`, when we have `CHECK_WRITE` below, correct?

Reply:
It is necessary due to the lazy initialization of memory.

Reply:
I think now `CHECK_WRITE` does do lazy memory initialization: https://github.com/nervosnetwork/ckb-vm/blob/1e600830cd9555bbdee19c9cd373065873362c09/src/machine/asm/execute_x64.S#L291, or am I missing something?

Reply:
Yes, you are right